pytroll_collectors/fsspec_to_message.py

```diff
@@ -25,31 +26,50 @@ def create_message(fs, file, subject, metadata=None):
 def _create_message_metadata(fs, file):
     """Create a message to send."""
     loaded_fs = json.loads(fs)
-    uri = _create_uri(file, loaded_fs)
-    uid = _create_uid_from_uri(uri, loaded_fs)
+    uid = _create_uid(file, loaded_fs)
+    uri = _create_uid_from_uri(uid, loaded_fs)
     base_data = {'filesystem': loaded_fs, 'uri': uri, 'uid': uid}
     base_data.update(file.get('metadata', dict()))
     return base_data
 
 
-def _create_uri(file, loaded_fs):
+def _create_uid(file, loaded_fs):
     protocol = loaded_fs["protocol"]
     if protocol == 'abstract' and 'zip' in loaded_fs['cls']:
         protocol = 'zip'
-    uri = protocol + '://' + file['name']
-    return uri
+    netloc = create_netloc(loaded_fs)
+    uid = protocol + '://' + netloc + file['name']
+    return uid
 
 
-def _create_uid_from_uri(uri, loaded_fs):
-    uid = uri
+def _create_uid_from_uri(uid, loaded_fs):
```

Comment:
This is creating URI from UID, so the naming is reversed

Reply:
Fixed